addons/calendar/static/tests/calendar_notification_tests.js

```diff
@@ -0,0 +1,205 @@
+    legacyServicesRegistry.add('bus_service', BusServiceOnce);
+    legacyServicesRegistry.add('local_storage', LocalStorageService);
+
+    testConfig = getActionManagerTestConfig();
+    testConfig.serviceRegistry.add('calendarNotification', calendarNotificationService);
+
+    testConfig.mainComponentRegistry = new Registry();
+    testConfig.mainComponentRegistry.add("NotificationContainer", NotificationContainer);
+
+    testConfig.browser.setTimeout = (fn) => fn();
+    testConfig.browser.clearTimeout = () => {};
+  });
+
+  QUnit.test('can listen on bus and displays notifications in DOM and click OK', async (assert) => {
+    assert.expect(5);
+
+    const mockRPC = (route, args) => {
+      if (route === '/longpolling/poll') {
+        return Promise.resolve([{
+          id: 'prout',
+          channel: ['db_name', 'calendar.alarm', 7],
+          message: [
+            {
+                "alarm_id": 1,
+                "event_id": 2,
+                "title": "Meeting",
+                "message": "Very old meeting message",
+                "timer": 20 * 60,
+                "notify_at": '1978-04-14 12:45:00',
+            }
+          ],
+        }]);
+      }
+      if (route === '/calendar/notify') {
+        return Promise.resolve(true);
+      }
+      if (route === '/calendar/notify_ack') {
+        assert.step("notifyAck");
+        return Promise.resolve(true);
+      }
+    };
+
+    const webClient = await createWebClient({
+      testConfig,
+      legacyParams: { serviceRegistry: legacyServicesRegistry },
+      mockRPC,
+    });
+
+    await nextTick();
+
+    assert.containsOnce(webClient.el, '.o_notification_body');
+    assert.strictEqual(
+      webClient.el.querySelector('.o_notification_body .o_notification_content').textContent,
+      "Very old meeting message"
+    );
+
+    await click(webClient.el.querySelectorAll('.o_notification_buttons .btn')[0]);
+    assert.verifySteps(["notifyAck"]);
+    assert.containsNone(webClient.el, ".o_notification");
+
+    webClient.destroy();
+  });
+
+  QUnit.test('can listen on bus and displays notifications in DOM and click Detail', async (assert) => {
+    assert.expect(5);
+
+    const mockRPC = (route, args) => {
+      if (route === '/longpolling/poll') {
+        return Promise.resolve([{
+          id: 'prout',
+          channel: ['db_name', 'calendar.alarm', 7],
+          message: [
+            {
+                "alarm_id": 1,
+                "event_id": 2,
+                "title": "Meeting",
+                "message": "Very old meeting message",
+                "timer": 20 * 60,
+                "notify_at": '1978-04-14 12:45:00',
+            }
+          ],
+        }]);
+      }
+      if (route === '/calendar/notify') {
+        return Promise.resolve(true);
+      }
+    };
+
+    const fakeActionService = {
+      name: "action",
+      deploy() {
+        return {
+          doAction(actionId) {
+            assert.step(actionId);
```

Comment:
you don't go here (??)

Reply:
It's verified that we go there on line 155 :     assert.verifySteps(["calendar.action_calendar_event_notify"]);